src/Adapter/MSTest.TestAdapter/Execution/TestExecutionManager.cs

```diff
@@ -174,6 +207,44 @@ public void RunTests(IEnumerable<string> sources, IRunContext? runContext, IFram
         }
     }
 
+    internal async Task RunTestsAsync(IEnumerable<string> sources, IRunContext? runContext, IFrameworkHandle frameworkHandle, TestRunCancellationToken cancellationToken)
```

Comment:
Note for review: This is an exact copy of the method above, but an async version.